client/directclient.go

```diff
@@ -117,7 +117,7 @@ func (c *ProxyDirectClient) firstResponse(reqs ...*http.Request) (resp *http.Res
 
 		select {
 		case resp = <-success:
-			if resp != nil && resp.StatusCode/100 == 2 {
+			if resp != nil && (resp.StatusCode/100 == 2 || resp.StatusCode == 304 || resp.StatusCode == 416) {
 				return resp
```

Comment:
IMHO, using status [constants from http pkg](https://golang.org/pkg/net/http/#pkg-constants) would be better.